tests/pcie/test_pcie_device.cpp

```diff
@@ -0,0 +1,52 @@
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <filesystem>
+#include <string>
+#include <vector>
+
+#include "device/pcie/pci_device.hpp"
+
+static std::vector<int> simple_pcie_device_enumeration()
```

Comment:
This is essentially what ttkmd_scan function does, right?
I'd really like for us to avoid duplication in some way.
Maybe both should be merged into a static PCIDevice::enumerate_devices ?